cli/upload.py

```diff
@@ -307,20 +309,37 @@ def _compose_file_mapping(upload_info: api.UploadInfo, xlsx: str):
             source_path = os.path.join(xlsx_dir, source_path)
 
             if not os.path.isfile(source_path):
-                missing_files.append(source_path)
+                if source_path in upload_info.optional_files:
+                    skipping_files.append(gcs_uri)
+                    continue
+                else:
+                    missing_files.append(source_path)
 
         # gsutil treats brackets in a gs-uri as a character set
         # see https://cloud.google.com/storage/docs/gsutil/addlhelp/WildcardNames#other-wildcard-characters
         # this replaces opening with a generic wildcard, which should map to only a single file
         elif "[" in source_path and "]" in source_path:
             source_path = source_path.replace("[", "?")
+            sub = subprocess.run(
+                ["gsutil", "ls", f"'{source_path}'"], capture_output=True
+            )
+            if sub.stdout.decode("utf-8").strip(" '").replace("[", "?") != source_path:
```

Comment:
Rather than inspecting stdout, you can rely on the fact that `sub.returncode == 0` if the file exists and `sub.returncode == 1` if not.

Reply:
good catch, forgot about returncode
in next commit